src/features/client/components/CartComponent/CartSideBar.jsx

```diff
@@ -0,0 +1,127 @@
+import {
+  Button,
+  Text,
+  Flex,
+  Heading,
+  Image,
+  Box,
+  Center,
+  Spacer,
+} from '@chakra-ui/react';
+import { useDispatch } from 'react-redux';
+import { changeAmount } from '../../../../store/reducers/cartReducer';
+
+function CartSideBar({ cartItems }) {
+  const dispatch = useDispatch();
+
+  const orderTotal = cartItems.length
+    ? cartItems
+        .reduce((acc, curr) => {
+          const currValue = +curr.recipePrice;
+          return acc + currValue * curr.quantity;
+        }, 0)
+        .toFixed(2)
+    : '0';
+
+  return (
+    <Flex height="100%" direction="column" position="relative">
+      <Box bg="white" height="400px" position="sticky" top="20px">
+        <Flex direction="column" height="100%">
+          <Center>
+            <Heading>Your order</Heading>
+          </Center>
+          <Box
+            padding="0px 20px"
+            mt="20px"
+            mb="20px"
+            maxHeight="300px"
+            overflow="scroll"
+            overflowX="hidden"
+          >
+            {cartItems.map((item, i) => {
+              const itemTotal = (+item.recipePrice * item.quantity).toFixed(2);
+              return (
+                <Flex
+                  key={i}
+                  marginBottom="10px"
+                  justifyContent="space-between"
+                >
+                  <Flex>
+                    <Center
+                      overflow="hidden"
+                      rounded="md"
+                      height="60px"
+                      marginRight="10px"
+                      minWidth="100px"
+                      width="30%"
+                      maxHeight="150px"
+                    >
+                      <Image src={item.recipeThumb} rounded="md" />
+                    </Center>
+                    <Center>
+                      <Box>
+                        <Heading fontSize="16px">{item.recipeName}</Heading>
+                        <Text fontSize="14px" marginTop="10px" color="blue.400">
+                          €{itemTotal}
+                        </Text>
+                      </Box>
+                    </Center>
+                  </Flex>
+                  <Flex alignItems="center" gap="5px">
+                    <Button
+                      onClick={() =>
```

Comment:
Move logic to map variable init space, pass function

Reply:
fixed